src/avisi/atlassian/connect/firestore.clj

```diff
@@ -0,0 +1,195 @@
+(ns avisi.atlassian.connect.firestore
+  (:require
+    [mount.core :as m :refer [defstate]]
+    [com.fulcrologic.guardrails.core :refer [>defn >def => ?]]
+    [clojure.spec.alpha :as s]
+    [camel-snake-kebab.core :as csk]
+    [avisi.atlassian.connect.jwt :as jwt]
+    [cambium.core :as log]
+    [clojure.string :as string])
+  (:import
+    (com.google.cloud.firestore
+      FirestoreOptions
+      FirestoreOptions$Builder
+      Firestore
+      DocumentReference
+      CollectionReference
+      DocumentSnapshot
+      Query
+      FieldPath
+      QueryDocumentSnapshot
+      QuerySnapshot)
+    (com.google.api.core ApiFuture)
+    (java.util HashMap ArrayList)
+    (java.time Duration)
+    (java.util Date UUID)
+    (com.google.cloud Timestamp)))
+
+(set! *warn-on-reflection* true)
+
+(>def ::collection-path string?)
+(>def ::document-path string?)
+(>def ::document map?)
+(>def ::host-client-key string?)
+(>def ::collection string?)
+(>def ::field string?)
+(>def ::value string?)
+
+(defn transform-map-keys [m f]
+  (reduce (fn [acc [k v]] (assoc acc (f k) (cond->> v (or (vector? v) (instance? ArrayList v)) (mapv f)))) {} m))
+
+(defn keys-as-string [m] (transform-map-keys m str))
+
+(defn string-keys-to-keywords [m]
+  (transform-map-keys
+    m
+    (fn [s]
+      (cond-> s
+        (string/starts-with? s ":")
```

Comment:
Ik vind de keywords opslaan zoals: `:foo.bar/foo` persoonlijk niet echt mooi, lijkt me ook erg lastig als je een keer iets wilt doen met de data in iets anders dan Clojure